### PR TITLE
Fix/fib run

### PR DESCRIPTION
**Pull request by oxlime**
Think that perhaps the issue is not putting the deduced operands into memory?

### CODE REVIEW

File: src/vm/core.zig [hunk 1]
```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
         return self.runInstruction(allocator, &instruction);
     }
 
+    pub fn insertDeducedOperands(self: *Self, allocator: Allocator, op: OperandsResult) !void {
+        try self.segments.memory.set(allocator, op.op_0_addr, op.op_0);
```
Comment (tcoratger) [inline src/vm/core.zig:260]: I think we must only do that if op0 was deducted: https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L369

Maybe we can implement the util logic https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L54-L74 in our `OperandsResult` structure wdt?

File: src/vm/core.zig [hunk 2]
```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
...
+        try self.segments.memory.set(allocator, op.op_1_addr, op.op_1);
```
Comment (tcoratger) [inline src/vm/core.zig:261]: Same as before but with op1 no?

File: src/vm/core.zig [hunk 3]
```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
...
+        try self.segments.memory.set(allocator, op.dst_addr, op.dst);
```
Comment (tcoratger) [inline src/vm/core.zig:262]: Same as before but with dest no?

File: src/vm/core.zig [hunk 4]
```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
         return self.runInstruction(allocator, &instruction);
     }
 
+    pub fn insertDeducedOperands(self: *Self, allocator: Allocator, op: OperandsResult) !void {
```
Comment (tcoratger) [inline src/vm/core.zig:259]: Can we add unit tests to confirm the logic here?

File: src/vm/core.zig (hunk 1 above)
Comment (oxlime) [reply src/vm/core.zig:260]: Are u thinking something like this 
https://github.com/oxlime/cairo-zig/blob/99173d5a81ddac28d6538d15609f0a59690432d8/src/vm/core.zig#L835

File: src/vm/core.zig (hunk 4 above)
Comment (oxlime) [reply src/vm/core.zig:263]: Let me know if we need more tests

**Review by tcoratger — APPROVED** [review]

**Pull request merged**